### DISCUSSION TITLE
codemagic.yaml not detected

### DISCUSSION THREAD

**Discussion by ahmedserag2** [codemagic-ci-cd/codemagic-docs#2901] [Q&A] [answered]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

My code repos are on a self-hosted gitlab instance. I am using yaml configurations and they worked like a charm for quite sometime. although when my pat token expired in gitlab i had to renew it but i had trouble detecting my codemagic.yaml config file. I had two branches dev and stage. dev took my a couple of refreshes and hitting the check for configfile button until it finally detected it. As for the stage branch there is no chance no matter how many times i refreshed it. i am guessing some sort of caching is enabled on the repos that somehow reads an old version ? i am not sure if anyone faced a similar issue please help.  

### Expected results

Expected to be able to detect my codemagic.yaml file. 

### Actual results

<img width="1412" alt="image" src="https://github.com/user-attachments/assets/d32367e9-a691-4a22-9eca-4bd348c716fe" />

**Comment by dam-ease** [comment, marked as answer]
Hello @ahmedserag2,
Please can you confirm that the yaml file is in the root directory on the stage branch as well?
Also,  if your repository is restricted to white-labeled ip addresses, please white-label the IP addresses below:
```
34.74.234.56/32
35.185.76.207/32
38.22.7.112/29
66.185.17.80/29
66.185.18.0/29
207.254.42.240/29
```
If the issue still persists, kindly share your Team ID here, you can get this by navigating to Teams > Team (Personal Account) > Team Settings, and copy from the url.
<img width="623" alt="Screenshot 2025-01-08 at 10 31 23" src="https://github.com/user-attachments/assets/2e40d00b-eb85-41c8-a0a2-5d33a5ea72d9" />
